cabal-testsuite/PackageTests/Check/IgnoreWarning/cabal.test.hs

```diff
@@ -0,0 +1,4 @@
+import Test.Cabal.Prelude
+
+-- Should ignore warnings if instructed so.
+main = cabalTest $ cabal "check" ["--ignore=ShortDesc"]
```

Comment:
Could we test some more scenarios here. 

* `--ignore` is passed an option which is not recognised, is it just ignored or fail?


Perhaps we should also enumerate all the possible errors here, so that if someone breaks the command line interface then we realise.

Reply:
As now, an unrecognised flag is simply ignored. I agree an “unrecognised ignore” warning would be useful.

The problem I am facing now is that, for it to happen, I think I need to make `CheckExplanationID` a proper sum type, with as many constructors as `CheckExplanation`.

It seems some code duplication, but I do not see any way around that.